### PR TITLE
Fix RCS1049: Simplify boolean comparison

### PR DESCRIPTION
**Pull request by xtqqczze**
https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1049.md

This PR includes changes to ComInterop that have also been made in dotnet/runtime#44564.

### CODE REVIEW

File: src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs [hunk 1]
```diff
@@ -846,7 +846,7 @@ internal PSModuleInfo LoadUsingMultiVersionModuleBase(string moduleBase, Manifes
 
                 // At this point, we haven't found an actual module, so try loading it as a
                 // PSSnapIn and then finally as an assembly in the GAC...
-                if ((found == false) && (moduleSpecification.Guid == null) && (moduleSpecification.Version == null) && (moduleSpecification.RequiredVersion == null) && (moduleSpecification.MaximumVersion == null))
+                if ((!found) && (moduleSpecification.Guid == null) && (moduleSpecification.Version == null) && (moduleSpecification.RequiredVersion == null) && (moduleSpecification.MaximumVersion == null))
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs:849]:
```suggestion
                if (!found && (moduleSpecification.Guid == null) && (moduleSpecification.Version == null) && (moduleSpecification.RequiredVersion == null) && (moduleSpecification.MaximumVersion == null))
```

File: src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs [hunk 2]
```diff
@@ -972,7 +972,7 @@ protected override void BeginProcessing()
                         // of this bug in Win8 where not responding can occur during data piping.
                         // We are reverting to Win7 behavior for {icm | icm} and {proxycommand | proxycommand}
                         // cases. For ICM | % ICM case, we are using remote steppable pipeline.
-                        if ((MyInvocation != null) && (MyInvocation.PipelinePosition == 1) && (MyInvocation.ExpectingInput == false))
+                        if ((MyInvocation != null) && (MyInvocation.PipelinePosition == 1) && (!MyInvocation.ExpectingInput))
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs:975]:
```suggestion
                        if ((MyInvocation != null) && (MyInvocation.PipelinePosition == 1) && !MyInvocation.ExpectingInput)
```

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
This PR makes changes to [ComInterop](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/ComInterop/README.md), so a corresponding PR has been opened: dotnet/runtime#44564

**Comment by xtqqczze** [comment]
dotnet/runtime#44564 is approved.

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
dotnet/runtime#44564 is merged.

**Pull request merged**